include/Options/Business/Nav_Menus.php

```diff
@@ -73,6 +83,65 @@ protected function get_data_structure(): array {
 		);
 	}
 
+	/**
+	 * Sanitizes option's value.
+	 * Can populate the `$errors` property with blocking and non-blocking errors: in case of non-blocking errors,
+	 * the value is sanitized and can be stored.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array   $value   Value to sanitize.
+	 * @param Options $options All options.
+	 * @return array|WP_Error The sanitized value. An instance of `WP_Error` in case of blocking error.
+	 *
+	 * @phpstan-return NavMenusValue|WP_Error
+	 */
+	protected function sanitize( $value, Options $options ) {
+		// Sanitize new value.
+		$value = parent::sanitize( $value, $options );
+
+		if ( is_wp_error( $value ) ) {
+			// Blocking error.
+			return $value;
+		}
+
+		/** @phpstan-var NavMenusValue $value */
+		if ( empty( $value ) ) {
+			// Nothing to validate.
+			return $value;
+		}
+
+		$all_langs      = array();
+		$language_terms = wp_list_pluck( $this->get_language_terms(), 'slug' );
+
+		foreach ( $value as $theme_slug => $menu_ids_by_location ) {
+			foreach ( $menu_ids_by_location as $location => $menu_ids ) {
+				// Make sure the language slugs correspond to an existing language.
+				$value[ $theme_slug ][ $location ] = array();
+
+				foreach ( $language_terms as $lang_slug ) {
+					if ( ! empty( $menu_ids[ $lang_slug ] ) ) {
+						$value[ $theme_slug ][ $location ][ $lang_slug ] = $menu_ids[ $lang_slug ];
+					}
+				}
+
+				// Detect unknown languages.
+				$all_langs = array_merge( $all_langs, $menu_ids );
+			}
+		}
+
+		/** @phpstan-var NavMenusValue $value */
+		$unknown_langs = array_diff_key( $all_langs, array_flip( $language_terms ) );
+
+		// Detect invalid language slugs.
+		if ( ! empty( $unknown_langs ) ) {
+			// Non-blocking error.
+			$this->add_unknown_languages_warning( array_keys( $unknown_langs ) );
+		}
```

Comment:
Instead of using directly `add_unknown_languages_warning()`, I suggest to create a new method in `Abstract_Option` which would remove unknown languages from an array and add the warning when some is encountered.
`function remove_unknown_languages( array $languages ): array`
I expect this to factorize more code with `Domains` as the query to get all languages terms would be done in this new method.

What do you think?

Reply:
As discussed, the `foreach` loops in `Domains` and `Nav_Menus` are too different:
- In `Domains` we fill new values, fallback to previous values, warn for missing languages.
- In `Nav_Menus` we only fill non-empty values.